fastly/client.go

```diff
@@ -651,8 +651,8 @@ func checkResp(resp *http.Response, err error) (*http.Response, error) {
 	}
 }
 
-// decodeBodyMap is used to decode an HTTP response body into a mapstructure struct.
-func decodeBodyMap(body io.Reader, out any) error {
+// DecodeBodyMap is used to decode an HTTP response body into a mapstructure struct.
+func DecodeBodyMap(body io.Reader, out any) error {
```

Comment:
Can you have the exported func call the non-exported func, to minimize the size of this diff?

Reply:
I could... or I could put this change to expose `decodeBodyMap` into a separate PR.

Reply:
This change (and other related changes) have now been moved to their own commit, which I'm maintaining in a local branch. Once this is ready, I'll push that set of changes in their own PR first.